### PR TITLE
[WIP] Docker Compose Added

### LINKED ISSUE #980: Use docker-compose to build and run app
It will reduce the steps to build and run the application from 3 to 1 command: `docker-compose up`
All the necessary port mapping, linking with couchdb and running the couchdb is right order will be handled in the docker-compose.yml file

Though one has to install docker-compose once, this would be a nice to have feature as on a daily basis one has to interact with only one command `docker-compose up`.

### PR DESCRIPTION
**Pull request by Anisha1234**
Fixes #980 

**Changes proposed in this pull request:**
- reduce the steps to build and run the application .
- Added ".dockerignore" file .
- Added "docker-compose.yml" for necessary port mapping, linking with couchdb and running the couchdb.
- Command to run docker-compose "docker-compose up".

**Work in progress**
* Compile ember app to use production environment

cc @jkleinsc @jglovier @tangollama

### CODE REVIEW

File: app/styles/_variables_mixins.scss [hunk 1]
```diff
@@ -21,6 +21,7 @@ $blue_lightest:         #eff2f5;
 $blue_lightest2:        #dee2e7;
 $red:                   #ff6d6f;
 $red_light:             rgba($red,.3);
+$red_lighter:           rgba($red,.75);
```
Comment (jkleinsc) [inline app/styles/_variables_mixins.scss]: These changes appear to be for another PR and shouldn't be in this one.

File: app/styles/components/_buttons.scss [hunk 2]
```diff
@@ -1,17 +1,17 @@
 .btn {
   &.on-white {
-    background-color: $blue_light;
-    color: $navy;
+    background-color: #00c4a2;
```
Comment (jkleinsc) [inline app/styles/components/_buttons.scss]: These changes also appear to be for a different PR.

**Comment by Anisha1234** [comment]
@jkleinsc I looked into hospitalrun-server and it appears that it does not have a existing Dockerfile. Can you guide me in creating one ?

**Comment by jkleinsc** [comment]
@Anisha1234 thanks for updating this PR. I am going to merge it in.  The hospitalrun-server work should be done in that repo.  I just checked and eHealth Africa has been working on this: https://github.com/eHealthAfrica/hospitalrun-server/tree/conf/docker
I believe they have a decent amount of the work done, so I'd prefer they finish what they have started there.
